streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -417,11 +277,11 @@ public void shouldSetRecordMetadata() {
 
         testDriver.pipeInput(consumerRecord1);
 
-        final List<Record> processedRecords = mockProcessors.get(0).processedRecords;
+        final List<String> processedRecords = supplier.theCapturedProcessor().processed;
```

Comment:
Seems to be a step backward to use `String` instead of `Record` -- also, `MockProcessor` actually also supports `KeyValueTimestamp` as type...